app/components/List/Item.tsx

```diff
@@ -47,22 +48,33 @@ const ListItem = (
     keyboardNavigation,
     ...rest
   }: Props,
-  ref?: React.Ref<HTMLAnchorElement>
+  ref: React.RefObject<HTMLAnchorElement>
 ) => {
   const theme = useTheme();
   const compact = !subtitle;
 
-  let itemRef: React.Ref<HTMLAnchorElement> =
+  let itemRef: React.RefObject<HTMLAnchorElement> =
     React.useRef<HTMLAnchorElement>(null);
   if (ref) {
     itemRef = ref;
   }
 
   const { focused, ...rovingTabIndex } = useRovingTabIndex(
-    itemRef as React.RefObject<HTMLAnchorElement>,
+    itemRef,
     keyboardNavigation || to ? false : true
   );
-  useFocusEffect(focused, itemRef as React.RefObject<HTMLAnchorElement>);
+  useFocusEffect(focused, itemRef);
+
+  const handleFocus = React.useCallback(() => {
+    if (itemRef.current) {
+      scrollIntoView(itemRef.current, {
+        scrollMode: "if-needed",
+        behavior: "auto",
+        block: "center",
+        boundary: window.document.body,
```

Comment:
Isn't this boundary the default?

Reply:
Doesn't look like. The scroll isn't prevented upon keyboard navigation if `boundary` is absent. This fixes the problem.